src/main/java/seedu/address/model/task/Task.java

```diff
@@ -37,4 +79,59 @@ public void markTask() {
     public void unmarkTask() {
         isDone = false;
     }
+
+    /**
+     * Returns true if task contains any of the description or deadline keywords.
+     * By default, empty lists will return true.
+     *
+     * @param descriptionKeywords Possibly empty list containing keywords for {@code Description}.
+     * @param deadlineKeywords Possibly empty list containing keywords for {@code Deadline}.
+     * @return boolean indicating if task contains supplied keywords.
+     */
+    public boolean containsKeywords(List<Description> descriptionKeywords, List<Deadline> deadlineKeywords) {
+        return (descriptionKeywords.isEmpty() || descriptionKeywords.contains(description))
+                && (deadlineKeywords.isEmpty() || deadlineKeywords.contains(deadline));
```

Comment:
Just checking, is this case-insensitive?

Reply:
hmmm, yes. will edit it to make it case-insensitive to be consistent with our find methods. Fields stored will be case-sensitive but find/contains will be case-insensitive